File: src/main/java/sharev/domain/card/repository/CardRepositoryCustom.java
package sharev.domain.card.repository;

import java.time.LocalDateTime;
import java.util.UUID;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import sharev.domain.card.dto.response.TempResponseCardDto;

public interface CardRepositoryCustom {

    Long searchConnectionCount(UUID gatheringId, Long profileId, LocalDateTime snapshotTime);

    Page<TempResponseCardDto> searchTempCards(UUID gatheringId, Long accountId, LocalDateTime snapshotTime,
                                              Pageable pageable);
}

File: src/main/kotlin/sharev/card/adapter/outbound/jpa/repository/CardRepositoryCustom.kt
package sharev.card.adapter.outbound.jpa.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import sharev.card.application.port.outbound.result.TempCard
import java.time.LocalDateTime
import java.util.*

fun interface CardRepositoryCustom {
    fun searchTempCards(
        gatheringId: UUID,
        myCardId: Long,
        snapshotTime: LocalDateTime,
        pageable: Pageable,
    ): Page<TempCard>
}
